kinesis/kinesis.go

```diff
@@ -610,6 +617,26 @@ func zlibCompress(data []byte) ([]byte, error) {
 	return b.Bytes(), nil
 }
 
+func gzipCompress(data []byte) ([]byte, error) {
+	var b bytes.Buffer
+
+	if data == nil {
+		return nil, fmt.Errorf("No data to compress.  'nil' value passed as data")
+	}
```

Comment:
Is this a valid case to return an error every time? I think we can return silently or print a debug message.

Reply:
I understand this was something we did earlier. However, I am not confident enough to say compression error is something which should be returned as FLB_ERROR at the end. Can you please confirm this would be only an error log and won't stop FluentBit? @PettitWesley

Reply:
I believe this will only cause one record to be discarded and the rest will continue, thus I do think we should log an error for this case.

See here: https://github.com/aws/amazon-kinesis-streams-for-fluent-bit/blob/mainline/kinesis/kinesis.go#L249

Reply:
Thanks.